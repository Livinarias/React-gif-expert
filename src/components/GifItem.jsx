import PropTypes from 'prop-types'

export const GifItem = ({title, url}) => {
  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GifItem.propType ={
  title: PropTypes.string.isRequired,
  url  : PropTypes.string.isRequired,
}
export default GifItem
