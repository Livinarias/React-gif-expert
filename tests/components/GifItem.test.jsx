import {render,screen} from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe('Pruebas en GifItem', () => {

    const title = "test";
    const url = "https://test.com/test.jpg";

  test('debe hacer match con snapshot ', () => {
    
    const {container} = render(<GifItem title={title} url={url}/>);
    expect(container).toMatchSnapshot();
  });

  test('debe mostrar la imagen con el url indicado y el ALT indicado', () => {
    
    render(<GifItem title={title} url={url}/>);
    // expect(screen.getByRole('img').src).toBe(url);
    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test('debe tener un titulo el componente', () => {
    
    render(<GifItem title={title} url={url}/>);
    expect(screen.getByText(title)).toBeTruthy();
  })
  
  
  
})
