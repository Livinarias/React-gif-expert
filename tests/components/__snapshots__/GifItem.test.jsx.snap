// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem debe hacer match con snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test"
      src="https://test.com/test.jpg"
    />
    <p>
      test
    </p>
  </div>
</div>
`;
